collect_app/src/androidTest/java/org/odk/collect/android/feature/smoke/BadServerTest.java

```diff
@@ -63,4 +64,29 @@ public void whenMediaFileHasMissingPrefix_showsAsUpdated() {
                 .clickGetBlankForm()
                 .assertText(R.string.newer_version_of_a_form_info);
     }
+
+    @Test
+    /*
+     A server that doesn't return hashes based on the md5 of the file for media files
+     would fool Collect into thinking there was a new one each time. Collect should still redownload
+     the file in this case (there's nothing else it can do), but it should only identify the form
+     as being updated if the file actually changed.
+
+     Open Rosa does not actually specify that the hash should be the literal md5 file hash, but
```

Comment:
I think this sentence is incorrect. https://docs.getodk.org/openrosa-form-list/#elements-within-mediafile